src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx

```diff
@@ -63,43 +61,34 @@ const LoginForm: React.FunctionComponent<PropsType> = ({
     try {
       await signIn(data);
 
-      if (venue.access?.includes(VenueAccessMode.Codes)) {
-        const result = await checkAccess({
-          venueId: venue.id,
-          code: data.code,
-        });
-        if (result.data === false) {
-          throw new Error("access denied");
-        }
-        console.log("result.data", result.data);
-        console.log(
-          "saving token to",
-          accessTokenKey(venue.id),
-          ": ",
-          result.data.token
-        );
-        setLocalStorageToken(venue.id, result.data.token);
+      let result = null;
+
+      switch (venue.access) {
+        case VenueAccessMode.Codes:
+          result = await checkAccess({
+            venueId: venue.id,
+            code: data.code,
+          });
+          break;
+        case VenueAccessMode.Emails:
+          result = await checkAccess({
+            venueId: venue.id,
+            email: data.email,
+          });
+          break;
+
+        default:
+          break;
       }
```

Comment:
Since the `checkAccess` function just calls `firebase.functions().httpsCallable("access-checkAccess")(data)`, and  that function is checking `if (isPasswordValid || isEmailValid || isCodeValid)`, there doesn't seem to be any reason for us to have this `switch` here at all.

Should be able to simplify this down even more I think.

Reply:
I'm not sure I get your point. This check here is using the `venue.access` string and decides which access exactly to check for. If I remove the switch then it will always check for both emails and codes which is not want we want.

Reply:
In each `case` of the switch we are running effectively the same code, where we don't need to.

When it is `VenueAccessMode.Codes || VenueAccessMode.Emails`, based on the logic I laid out in https://github.com/sparkletown/sparkle/pull/1016#discussion_r565063616, the following code will make the correct check regardless:

(forgive the formatting here, prettier would make it nicer looking)
```typescript
const accessGranted = [VenueAccessMode.Codes, VenueAccessMode.Emails].includes(venue.access) ?
    await checkAccess({
      venueId: venue.id,
      email: data.email,
      code: data.code,
}) : true;
```

This way we don't need the `switch`/`case`, the mutable variable, or the code duplication.